SMLHelper/BiomeThings/BiomeClass.cs

```diff
@@ -0,0 +1,107 @@
+using UWE;
+using WorldStreaming;
+using System.Runtime.CompilerServices;
+namespace SMLHelper.V2.BiomeThings
+{
+    ///    <summary>
+    /// A class for adding biomes.
+    ///   </summary>
+    public abstract class Biome
+    {
+        /// <summary>
+        /// The sky of your biome.
+        /// </summary>
+        public abstract mset.Sky Sky { get; }
+        /// <summary>
+        /// The Waterscape Volume Settings of your biome.
+        /// </summary>
+        public abstract WaterscapeVolume.Settings WaterScapeSettings { get; }
+        /// <summary>
+        /// SpawnInfos for what you want to initially spawn into your biome without initializing a coordinated spawn class.
+        /// </summary>
+        public abstract List<SMLHelper.V2.Handlers.SpawnInfo> SpawnInfos { get; }
+        
+        /// <summary>
+        /// The name of the added biome, must be a string.
+        /// </summary>
+        public abstract string BiomeName { get; }
+        /// <summary>
+        /// The Rich Presence Message of the biome, must be a string.
+        /// </summary>
+        public virtual string BiomeRichPresence { get; }
+        /// <summary>
+        /// The Ambient Light Settings of your biome.
+        /// </summary>
+        public abstract AmbientLightSettings amblightsettings { get; }
+        /// <summary>
+        /// The Fog Settings for the biome.
+        /// </summary>
+        public abstract FogSettings fogsettings { get; }
+        /// <summary>
+        /// Sunlight Settings for the biome.
+        /// </summary>
+        public abstract SunlightSettings sunsettings { get; }
+        /// <summary>
+        /// The batch Ids, of the biome, must be the same as the ones in your .optoctrees files
+        /// </summary>
+        public abstract List<Int3> batchIds { get; }
+        internal Dictionary<Int3, GameObject> batchroots = new Dictionary<Int3, GameObject>();
+        /// <summary>
+        /// Call this method to Finialize setting values, and add the biome to the game.
+        /// </summary>
+        public void Patch()
+        { 
+           
+            Variables.biomes.Add(this);
+            
+            QModManager.Utility.Logger.Log(QModManager.Utility.Logger.Level.Info, $"Patched biome {BiomeName}");
+        }
+        /// <summary>
+        /// The BiomeMap you have edited.
+        /// </summary>
+        /// <returns>An Array of Color32</returns>
+        public abstract Texture2D EditedBiomeMap(out int height, out int width);
+        /// <summary>
+        /// The gameobject with the collider for registering when biome is entered
+        /// </summary>
+        /// <returns>A gameobject with collider applied</returns>
+        public abstract GameObject GetCollider();
+        /// <summary>
+        /// The Terrain for each batch of your biome, in format (BatchId,Terrain)
+        /// </summary>
+        public abstract Dictionary<Int3, GameObject> batchTerrains { get; }
```

Comment:
All `public`, `protected, and `internal` members should be in _PascalCase_.

Use _camelCase_ only for `private` members